ui/qt-lib/src/Ariadne/UI/Qt/Widgets/Dialogs/InsertPassword.hs

```diff
@@ -0,0 +1,153 @@
+module Ariadne.UI.Qt.Widgets.Dialogs.InsertPassword
+  ( runInsertPassword
+  , InsertPasswordResult(..)
+  ) where
+
+import Universum
+
+import Control.Lens (makeLensesWith)
+
+import Data.Bits
+import qualified Data.Text as T
+
+import Graphics.UI.Qtah.Core.Types (alignHCenter, alignVCenter)
+import Graphics.UI.Qtah.Signal (connect_)
+import Graphics.UI.Qtah.Widgets.QSizePolicy (QSizePolicyPolicy(..))
+
+import qualified Graphics.UI.Qtah.Core.QObject as QObject
+import qualified Graphics.UI.Qtah.Gui.QIcon as QIcon
+import qualified Graphics.UI.Qtah.Widgets.QAbstractButton as QAbstractButton
+import qualified Graphics.UI.Qtah.Widgets.QBoxLayout as QBoxLayout
+import qualified Graphics.UI.Qtah.Widgets.QCheckBox as QCheckBox
+import qualified Graphics.UI.Qtah.Widgets.QDialog as QDialog
+import qualified Graphics.UI.Qtah.Widgets.QHBoxLayout as QHBoxLayout
+import qualified Graphics.UI.Qtah.Widgets.QLabel as QLabel
+import qualified Graphics.UI.Qtah.Widgets.QLayout as QLayout
+import qualified Graphics.UI.Qtah.Widgets.QLineEdit as QLineEdit
+import qualified Graphics.UI.Qtah.Widgets.QPushButton as QPushButton
+import qualified Graphics.UI.Qtah.Widgets.QWidget as QWidget
+
+import Ariadne.UI.Qt.UI
+import Ariadne.UI.Qt.Widgets.Dialogs.Util
+import Ariadne.Util
+
+data InsertPassword = InsertPassword
+    { insertPassword :: QDialog.QDialog
+    , hasPassword :: QCheckBox.QCheckBox
+    , password :: QLineEdit.QLineEdit
+    , passwordWidget :: QWidget.QWidget
+    , confirmButton :: QPushButton.QPushButton
+    }
+
+data InsertPasswordResult
+    = InsertPasswordCanceled
+    | InsertPasswordAccepted (Maybe Text)
+
+makeLensesWith postfixLFields ''InsertPassword
+
+createPasswordField :: Text -> IO (QHBoxLayout.QHBoxLayout, QLineEdit.QLineEdit)
+createPasswordField placeholder = do
+    layout <- QHBoxLayout.new
+    field <- QLineEdit.new
+    QLineEdit.setEchoMode field QLineEdit.Password
+    QLineEdit.setPlaceholderText field $ toString placeholder
+
+    visibleButton <- QPushButton.new
+    void $ setProperty visibleButton ("styleRole" :: Text) ("passwordVisibilityToggle" :: Text)
```

Comment:
This style (`passwordVisibilityToggle`) is applied only to buttons in `#newWalletDialog`. I believe you should change QSS to `QDialog QPushButton[styleRole="passwordVisibilityToggle"]` (see `ui/qt-lib/resources/stylesheet.qss`)